mapper/mapper.go

```diff
@@ -2,13 +2,86 @@ package mapper
 
 import (
 	"context"
+	"fmt"
+	"strconv"
 
-	"github.com/ONSdigital/dp-frontend-models/model/homepage"
+	"github.com/ONSdigital/dp-api-clients-go/zebedee"
+	model "github.com/ONSdigital/dp-frontend-models/model/homepage"
+	"github.com/ONSdigital/log.go/log"
 )
 
 // Homepage maps data to our homepage frontend model
-func Homepage(ctx context.Context) homepage.Page {
-	var page homepage.Page
+func Homepage(ctx context.Context, mainFigures []model.MainFigure) model.Page {
+	var page model.Page
 	page.Type = "homepage"
+	page.Metadata.Title = "Home"
+	page.Data.MainFigures = mainFigures
 	return page
 }
+
+// MainFigure maps a single main figure object
+func MainFigure(ctx context.Context, datePeriod string, data zebedee.TimeseriesMainFigure) model.MainFigure {
+	var mf model.MainFigure
+
+	mfData := getDataByPeriod(datePeriod, data)
+	latestData := mfData[(len(mfData) - 1)]
+	previousData := mfData[(len(mfData) - 2)]
+	latestFigure, err := strconv.ParseFloat(latestData.Value, 64)
+	if err != nil {
+		log.Event(ctx, "error getting trend description: error parsing float", log.Error(err))
+		return mf
+	}
+	previousFigure, err := strconv.ParseFloat(previousData.Value, 64)
+	if err != nil {
+		log.Event(ctx, "error getting trend description: error parsing float", log.Error(err))
+		return mf
+	}
+
+	mf.Figure = fmt.Sprintf("%0.1f", latestFigure)
```

Comment:
Putting comment here as likely most relevant, can we have the controller format the figure UK population should display at xx,xxx,xxx but atm gets to the renderer as xxxxxxxx.x (renderer technically can do it but this is logic that I think should be handled on the controller - else renderer will get bloated with logic)

Reply:
👍